var/spack/repos/builtin/packages/postgis/package.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Postgis(AutotoolsPackage):
+    """
+    PostGIS is a spatial database extender for PostgreSQL object-relational
+    database. It adds support for geographic objects allowing location
+    queries to be run in SQL
+    """
+
+    homepage = "https://postgis.net/"
+    url      = "https://download.osgeo.org/postgis/source/postgis-2.5.3.tar.gz"
+
+    version('3.0.1', sha256='5a5432f95150d9bae9215c6d1c7bb354e060482a7c379daa9b8384e1d03e6353')
+    version('3.0.0', sha256='c06fd2cd5cea0119106ffe17a7235d893c2bbe6f4b63c8617c767630973ba594')
+    version('2.5.3', sha256='72e8269d40f981e22fb2b78d3ff292338e69a4f5166e481a77b015e1d34e559a')
+
+    variant('gui', default=False, description='Build with GUI support, creating shp2pgsql-gui graphical interface to shp2pgsql')
+
+    # Refs:
+    # https://postgis.net/docs/postgis_installation.html
+    # https://postgis.net/source/
+
+    depends_on('postgresql')
+    depends_on('geos')
+    depends_on('proj')
+    depends_on('gdal')
+    depends_on('libxml2')
+    depends_on('json-c')
+
+    depends_on('sfcgal')
+    depends_on('pcre')
+    depends_on('perl')
+    depends_on('protobuf-c')
+
+    depends_on('gtkplus@:2.24.32', when='+gui')
+
+    def setup_build_environment(self, env):
+        env.set('POSTGIS_GDAL_ENABLED_DRIVERS', 'ENABLE_ALL')
+
+    def setup_run_environment(self, env):
+        env.set('POSTGIS_GDAL_ENABLED_DRIVERS', 'ENABLE_ALL')
+
+    def configure_args(self):
+        args = []
+        args.append('--with-sfcgal=' + str(self.spec['sfcgal'].prefix.bin) +
+                    '/sfcgal-config')
+        if '+gui' in self.spec:
+            args.append('--with-gui')
```

Comment:
Can we explicitly disable this in an else clause?

Reply:
I tried passing `--without-gui` which in fact ended up enabling it! :confused: 
```
sbulut@ws-067 /cache/sbulut/spack-stage/spack-stage-postgis-3.0.1-2o2utebs4svr5rx7eswmhggafcqczgvq/spack-src 
$ ./configure
...
$ grep -E 'gui|GTK' config.log
GTK_BUILD=''
GTK_CFLAGS=''
GTK_LIBS=''
GTK_WIN32_FLAGS=''
GTK_WIN32_RES=''
```
Now with without-gui we get:
```
$ ./configure --without-gui
...
$ grep -E 'gui|GTK' config.log
  $ ./configure --without-gui
configure:17978: result: GUI: Build requested, checking for dependencies (GKT+2.0)
configure:18055: checking for GTK+ - version >= 2.8.0
configure:18165: /mnt/local/spack/lib/spack/env/gcc/gcc -o conftest -std=gnu99 -g -O2 -fno-math-errno -fno-signed-zeros -pthread -I/mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-8.3.0/pcre-8.44-y4f6e6ifxdrku4kwgjvj6xtwi6yj4c45/include -I/mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-8.3.0/zlib-1.2.11-i2s36g33dqf25qmj4lnggjbaariws7ui/include -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng15 -I/usr/include/uuid -I/usr/include/pixman-1 -I/usr/include/libdrm     conftest.c -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgio-2.0 -lpangoft2-1.0 -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lcairo -lpango-1.0 -lfontconfig -lgobject-2.0 -lglib-2.0 -lfreetype    >&5
GTK_BUILD='gui'
GTK_CFLAGS='-pthread -I/mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-8.3.0/pcre-8.44-y4f6e6ifxdrku4kwgjvj6xtwi6yj4c45/include -I/mnt/local/spack/opt/spack/linux-centos7-skylake/gcc-8.3.0/zlib-1.2.11-i2s36g33dqf25qmj4lnggjbaariws7ui/include -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng15 -I/usr/include/uuid -I/usr/include/pixman-1 -I/usr/include/libdrm  '
GTK_LIBS='-lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgio-2.0 -lpangoft2-1.0 -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lcairo -lpango-1.0 -lfontconfig -lgobject-2.0 -lglib-2.0 -lfreetype  '
GTK_WIN32_FLAGS=''
GTK_WIN32_RES=''
```
So keep it as it is I guess?